ui/src/tool-panel/switch/dialog/geometry-switch-infobox-container.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import GeometrySwitchInfobox from 'tool-panel/switch/geometry-switch-infobox';
+import { useCommonDataAppSelector, useTrackLayoutAppSelector } from 'store/hooks';
+import { createDelegates } from 'store/store-utils';
+import {
+    GeometrySwitchInfoboxVisibilities,
+    trackLayoutActionCreators as TrackLayoutActions,
+} from 'track-layout/track-layout-slice';
+import { LayoutSwitch } from 'track-layout/track-layout-model';
+import { Point } from 'model/geometry';
+import { calculateBoundingBoxToShowAroundLocation } from 'map/map-utils';
+import { GeometryPlanId, GeometrySwitchId } from 'geometry/geometry-model';
+
+type GeometrySwitchInfoboxContainerProps = {
+    onVisibilityChange: (visibilities: GeometrySwitchInfoboxVisibilities) => void;
+    visibilities: GeometrySwitchInfoboxVisibilities;
+    switchId: GeometrySwitchId;
+    layoutSwitch: LayoutSwitch | undefined;
+    planId: GeometryPlanId;
+};
+
+export const GeometrySwitchInfoboxContainer: React.FC<GeometrySwitchInfoboxContainerProps> = ({
+    switchId,
+    layoutSwitch,
+    planId,
+    visibilities,
+    onVisibilityChange,
+}) => {
+    const trackLayoutState = useTrackLayoutAppSelector((state) => state);
+    const changeTimes = useCommonDataAppSelector((state) => state.changeTimes);
+    const delegates = React.useMemo(() => createDelegates(TrackLayoutActions), []);
+
+    const onShowMapLocation = React.useCallback(
+        (location: Point) => delegates.showArea(calculateBoundingBoxToShowAroundLocation(location)),
+        [],
+    );
```

Comment:
Hmm. Osassa paikoista passailtiin kartankohdistusfunkkarina lambdafunkkarireferenssi, kun taas näytettäisiin määriteltävän sama toiminnallisuus useCallbackilla? Pitäiskö muissakin paikoissa cachettaa kartankohdistusfunkkari tällä tavoin (varmaan perffin puolesta tosin aika minimaalinen hyöty)?

Reply:
Tuo siirtyi varmaan jostain alempaa ketjusta, ja siellä aikanaan tosiaan koitettiin pitää uusien asioiden luonnit minimissään perffin parantamiseksi. En usko että tässä tuolla on juurikaan väliä kun nämä ovat jo memotetun `ToolPanel`:in sisällä, niin taidan poistaa nuo callbackisaatiot täältä ja sieltä toisesta paikasta